app/models/pipeline_run.rb

```diff
@@ -210,6 +573,22 @@ def file_generated_since_run(s3_path)
     end
   end
 
+  def file_generated_since_jobstats?(s3_path)
+    # If there is no file, return false
+    stdout, _stderr, status = Open3.capture3("aws", "s3", "ls", s3_path.to_s)
+    return false unless status.exitstatus.zero?
+    # If there is a file and there are no existing job_stats yet, return true
+    existing_jobstats = job_stats
+    return true if existing_jobstats.empty?
```

Comment:
Why not crash?

Reply:
job_stats isn't an explicit argument to this function, probably should be

Reply:
I don't see the point in giving it job_stats as an argument. They are being treated as properties of the pipeline_run.